scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -0,0 +1,417 @@
+    {
+      if (bracket1 < dxLow || bracket1 > dxHigh || bracket2 < dxLow || bracket2 > dxHigh) {
+        throw new WrongBracketingException("Brackets are out of prescribed root domain");
+      }
+      final var ff = new Function<Double>(){
+      @Override
+      public Double valueAt(final Double x) {
+        return f.valueAt(x) - y;
+      }
+
+      @Override
+      public boolean isApproximation() {
+        return f.isApproximation();
+      }
+    };
+
+      double fx_m = ff.valueAt(bracket1);
+      double fx_p = ff.valueAt(bracket2);
+      if (fx_m * fx_p > 0) {
+        throw new WrongBracketingException("f(x0)=" + fx_m + " and f(x1)=" + fx_p + " are of the same sign");
+      }
+      for (int nbIt = 0; nbIt < maxNbIterations; nbIt++) {
+        final double xg = (bracket1 + bracket2) / 2.;
+        final var fxg = ff.valueAt(xg);
+        if (fxg >= confidenceYLow && fxg <= confidenceYHigh) {
+          return new RootFindingResult<>(xg, fxg,null);
+        } else {
+          if (fxg > 0 && bracket1 > 0 || fxg < 0 && bracket1 < 0) {
+            bracket1 = xg;
+          } else {
+            bracket2 = xg;
+          }
+        }
+
+      }
+      throw new ExceededMaxIterationException();
+    }
+
+  }
+
+  final static class BisectionDurationAlgorithm implements BracketingAlgorithm<Duration> {
+
+    public RootFindingResult<Duration> findRoot(
+        Function<Duration> f,
+        Duration bracket1,
+        Duration bracket2,
+        Duration y,
+        Duration confidenceYLow,
+        Duration confidenceYHigh,
+        Duration dxLow,
+        Duration dxHigh,
+        int maxNbIterations)
+    throws WrongBracketingException, ExceededMaxIterationException
+    {
+      if (bracket1.shorterThan(dxLow) || bracket1.longerThan(dxHigh) || bracket2.shorterThan(dxLow) || bracket2.longerThan(dxHigh)) {
+        throw new WrongBracketingException("Brackets are out of prescribed root domain");
+      }
+      final var ff = new Function<Duration>(){
+        @Override
+        public Duration valueAt(final Duration x) {
+          return f.valueAt(x).minus(y);
+        }
+
+        @Override
+        public boolean isApproximation() {
+          return f.isApproximation();
+        }
+      };
+      var bracket1Long = bracket1.in(Duration.MICROSECONDS);
+      var bracket2Long = bracket2.in(Duration.MICROSECONDS);
+      var fx_m = ff.valueAt(bracket1);
+      var fx_p = ff.valueAt(bracket2);
+      if (fx_m.in(Duration.MICROSECONDS) * fx_p.in(Duration.MICROSECONDS) > 0) {
+        throw new WrongBracketingException("f(bracket1) and f(bracket2) are of the same sign");
+      }
+      for (int nbIt = 0; nbIt < maxNbIterations; nbIt++) {
+        long xg = Math.round(((float) bracket1Long + bracket2Long) / 2);
+        final var fxg = ff.valueAt(Duration.of(xg, Duration.MICROSECONDS));
+        if (fxg.noShorterThan(confidenceYLow) && fxg.noLongerThan(confidenceYHigh)) {
```

Comment:
Similar question as above - is `yLow` expected to be a positive, or a negative number? If it's positive, I would expect this to be something like this:
```suggestion
        if (fxg.noShorterThan(Duration.negate(confidenceYLow)) && fxg.noLongerThan(confidenceYHigh)) {
```

Reply:
This was unsure and lead to this error. They should be positive.